scripts/ice-road-pipeline.py

```diff
@@ -124,15 +164,170 @@
+            lines = f.readlines()
+            for line in lines:
+                if "Translation vector (North-East-Down, meters): Vector3" in line: 
+                    n_e_d_shift = re.findall("[+-]?\d+\.\d+", line)
+                    n_e_d_shift = ','.join(n_e_d_shift)
+                    break
+
+        # get crs
+        ref_raster = rasterio.open(snow_tif)
+        crs = ref_raster.crs
+        pix_size, _ = ref_raster.res
+        crs = str(crs).split(":",1)[1]    
+        ras_meta = ref_raster.profile
+
+        # Compute slope and aspect from snow-on lidar
+        slope_fp = join(ice_dir, f'{basename(in_dir)}-snowon_slope.tif')
+        aspect_fp = join(ice_dir, f'{basename(in_dir)}-snowon_aspect.tif')
+        os.system(f'gdaldem slope -compute_edges {snow_tif} {slope_fp} -q')
+        os.system(f'gdaldem aspect -compute_edges -zero_for_flat {snow_tif} {aspect_fp} -q')
+        slope = rio.open_rasterio(slope_fp, masked=True)
+        aspect = rio.open_rasterio(aspect_fp, masked=True)
+
+        # Compute surface normal vector from snow-on lidar
+        ni_fp = join(ice_dir, f'{basename(in_dir)}-snowon_n_i.tif')
+        nj_fp = join(ice_dir, f'{basename(in_dir)}-snowon_n_j.tif')
+        nk_fp = join(ice_dir, f'{basename(in_dir)}-snowon_n_k.tif')
+        n_i = np.sin(np.radians(aspect)) * np.sin(np.radians(slope))
+        n_j = np.cos(np.radians(aspect)) * np.sin(np.radians(slope)) 
+        n_k = np.cos(np.radians(slope))
+        n_i.rio.to_raster(ni_fp)
+        n_j.rio.to_raster(nj_fp)
+        n_k.rio.to_raster(nk_fp)        
+        
+        # Calcs calibration stats from target
+        output_csv = f'{ssa_dir}/all-calibration-rfl.csv'
+        subprocess.call(["Rscript", 
+                          f"{scripts_dir}/las_ssa_cal.r", 
+                          cal_las, crs, shp_fp_rfl, n_e_d_shift, 
+                          output_csv, imu_data, str(pix_size)])
+        
+        # Read in cal data and estimate factor
+        df = pd.read_csv(output_csv)
+        median_rfl = df['rfl'].median()
+        road_cal_factor = known_rfl / median_rfl
+        road_cal_factor = str(road_cal_factor) #convert to str for R
+
+        # For each file in <in-dir> 
+        for f in os.listdir(in_dir):
+             
+            # Set names
+            base_las = os.path.basename(f)
+            las_name = os.path.splitext(base_las)[0]
+
+            # R is making a .DS_Store in in_dir, and I can't figure out how or where...
+            # So this will be just a quick check to continue to next iter if encountered.
+            if las_name == ".DS_Store" or las_name == "ice-road":
+                continue 
+
+            rfl_fp = f'{ssa_dir}/{las_name}-rfl.las'
+
+            # Getting a translated "LAS" file
+            # "LAS" in quotations bc I am hiding the rfl here in "Z"
+            # with the intention to do fast IDW in the next step.
+            if os.path.exists(rfl_fp):
```

Comment:
Any reason we would want to over-write these files? If this step doesn't take too long it might make sense to just overwrite the existing files so we can iterate if we change things? Or add a check if the debug flag is on to re-create these files?

Reply:
That is a good point, this was in there for me during testing. I can't really see a purpose other than for testing the code that we would actually want to use this... 

I will remove it for now, and if it ends up being an issue we can add the ability for the code to print out a user option to overwrite or keep existing and pass.